internal/ethapi/api.go

```diff
@@ -724,11 +728,42 @@ func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, number rpc.B
 func (s *PublicBlockChainAPI) GetBlockByHash(ctx context.Context, hash common.Hash, fullTx bool) (map[string]interface{}, error) {
 	block, err := s.b.BlockByHash(ctx, hash)
 	if block != nil {
-		return s.rpcMarshalBlock(ctx, block, true, fullTx)
+		result, err := s.rpcMarshalBlock(ctx, block, true, fullTx)
+		if err != nil {
+			return nil, err
+		}
+		addEthCompatibilityFields(ctx, result, s.b, block.Header())
+		return result, nil
 	}
 	return nil, err
 }
 
+// addEthCompatibilityFields seeks to work around the incompatibility of celo
+// and ethers.js (and potentially other web3 clients) by adding fields to our
+// rpc response that ethers.js depends upon.
+// See https://github.com/celo-org/celo-blockchain/issues/1945
+func addEthCompatibilityFields(ctx context.Context, block map[string]interface{}, b Backend, header *types.Header) {
+	hash := header.Hash()
+	numhash := rpc.BlockNumberOrHash{
+		BlockHash: &hash,
+	}
+	gasLimit := b.GetBlockGasLimit(ctx, numhash)
+	block["gasLimit"] = hexutil.Uint64(gasLimit)
+
+	// Providing nil as the currency address gets the gas price minimum for the native celo asset.
+	baseFee, err := b.GasPriceMinimumForHeader(ctx, nil, header)
```

Comment:
Should we also mock the gasPriceMinimum for blocks that we don't have the state?
The same as we are doing with the `GetBlockGasLimit` where we are returning `params.DefaultGasLimit` which is set to 20M.
Both have different natures and the blockGasLimit will fluctuate only by governance (which is not that common), and the price is tied to usage. But we are already sending a possible "wrong" answer. Should we do the same with the `baseFee`? Or maybe we should put a nil in the gasLimit too if we don't have the state

Reply:
Same as the other comments, sorry for didn't check that before.
I agree that we should be consistent. Or we return null in both cases, or we return a mocked things

Reply:
I'm leaning towards no mocking, because if we switch to actually putting these fields on the block then there will be a point when the fields don't exist. Otherwise we have to maintain this shim forever. But I'm really not sure.